sparql/SPARQLLexer.g4

```diff
@@ -0,0 +1,301 @@
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * (REVISION Last updated: 18 October 2023)
+ */
+lexer grammar SPARQLLexer;
+
+options { caseInsensitive = true; }
+
+SPACE : WS -> channel(HIDDEN);
+COMMENT : '#' ~[\r\n]* -> channel(HIDDEN);
+
+TRUE : 'TRUE';
+FALSE : 'FALSE';
+DISTINCT : 'DISTINCT';
+NOT : 'NOT';
+IN : 'IN';
+STR : 'STR';
+LANG : 'LANG';
+LANGMATCHES : 'LANGMATCHES';
+DATATYPE : 'DATATYPE';
+BOUND : 'BOUND';
+IRI : 'IRI';
+URI : 'URI';
+BNODE : 'BNODE';
+RAND : 'RAND';
+ABS : 'ABS';
+CEIL : 'CEIL';
+FLOOR : 'FLOOR';
+ROUND : 'ROUND';
+CONCAT : 'CONCAT';
+STRLEN : 'STRLEN';
+UCASE : 'UCASE';
+LCASE : 'LCASE';
+ENCODE_FOR_URI : 'ENCODE_FOR_URI';
+CONTAINS : 'CONTAINS';
+STRSTARTS : 'STRSTARTS';
+STRENDS : 'STRENDS';
+STRBEFORE : 'STRBEFORE';
+STRAFTER : 'STRAFTER';
+YEAR : 'YEAR';
+MONTH : 'MONTH';
+DAY : 'DAY';
+HOURS : 'HOURS';
+MINUTES : 'MINUTES';
+SECONDS : 'SECONDS';
+TIMEZONE : 'TIMEZONE';
+TZ : 'TZ';
+NOW : 'NOW';
+UUID : 'UUID';
+STRUUID : 'STRUUID';
+MD5 : 'MD5';
+SHA1 : 'SHA1';
+SHA256 : 'SHA256';
+SHA384 : 'SHA384';
+SHA512 : 'SHA512';
+COALESCE : 'COALESCE';
+IF : 'IF';
+STRLANG : 'STRLANG';
+STRDT : 'STRDT';
+SameTerm : 'SAMETERM';
+IsIRI : 'ISIRI';
+IsURI : 'ISURI';
+IsBLANK : 'ISBLANK';
+IsLITERAL : 'ISLITERAL';
+IsNUMERIC : 'ISNUMERIC';
+COUNT : 'COUNT';
+SUM : 'SUM';
+MIN : 'MIN';
+MAX : 'MAX';
+AVG : 'AVG';
+SAMPLE : 'SAMPLE';
+GROUP_CONCAT : 'GROUP_CONCAT';
+SEPARATOR : 'SEPARATOR';
+REGEX : 'REGEX';
+SUBSTR : 'SUBSTR';
+REPLACE : 'REPLACE';
+EXISTS : 'EXISTS';
+SELECT : 'SELECT';
+REDUCED : 'REDUCED';
+AS : 'AS';
+WHERE : 'WHERE';
+OPTIONAL : 'OPTIONAL';
+GRAPH : 'GRAPH';
+SERVICE : 'SERVICE';
+SILENT : 'SILENT';
+BIND : 'BIND';
+VALUES : 'VALUES';
+UNDEF : 'UNDEF';
+A_ : 'A';
+UNION : 'UNION';
+MINUS : 'MINUS';
+FILTER : 'FILTER';
+GROUP : 'GROUP';
+BY : 'BY';
+HAVING : 'HAVING';
+ASC : 'ASC';
+DESC : 'DESC';
+LIMIT : 'LIMIT';
+OFFSET : 'OFFSET';
+ORDER : 'ORDER';
+DEFAULT : 'DEFAULT';
+NAMED : 'NAMED';
+ALL : 'ALL';
+USING : 'USING';
+INSERT : 'INSERT';
+DELETE : 'DELETE';
+WITH : 'WITH';
+DATA : 'DATA';
+COPY : 'COPY';
+TO : 'TO';
+MOVE : 'MOVE';
+CREATE : 'CREATE';
+LOAD : 'LOAD';
+INTO : 'INTO';
+CLEAR : 'CLEAR';
+DROP : 'DROP';
+BASE : 'BASE';
+PREFIX : 'PREFIX';
+CONSTRUCT : 'CONSTRUCT';
+FROM : 'FROM';
+DESCRIBE : 'DESCRIBE';
+ASK : 'ASK';
+
+CARET2 : '^^';
+CARET : '^';
+OPAR : '(';
+CPAR : ')';
+COMMA : ',';
+OR : '||';
+AND : '&&';
+EQ : '=';
+NEQ : '!=';
+GT : '>';
+LT : '<';
+GTE : '>=';
+LTE : '<=';
+ADD : '+';
+SUB : '-';
+MUL : '*';
+DIV : '/';
+EXCL : '!';
+SCOL : ';';
+OBRACE : '{';
+CBRACE : '}';
+DOT : '.';
+OBRACK : '[';
+CBRACK : ']';
+PIPE : '|';
+QMARK : '?';
+
+/// [139]    IRIREF : '<' ([^<>"{}|^`\]-[#x00-#x20])* '>'
```

Comment:
I'd remove all comments to tokens because they look obvious and don't clarify anything.